File: hardware-mall-backend/src/main/java/com/example/mystore/entity/db/Order.java
package com.example.mystore.entity.db;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("shop_order")
public class Order {
    @TableId(type = IdType.AUTO)
    private Long id;
    
    private String orderNo;
    
    private Long userId;
    
    private Long addressId;
    
    private Long logisticsId;
    
    private Integer status;
    
    private BigDecimal totalAmount;

    private BigDecimal freightAmount;

    private BigDecimal payAmount;
    
    private String logisticsNo;
    
    private LocalDateTime payTime;
    
    private LocalDateTime shipTime;
    
    private LocalDateTime receiveTime;
    
    private LocalDateTime cancelTime;
    
    private String cancelReason;
    
    private String buyerRemark;
    
    private String adminRemark;
    
    private LocalDateTime createTime;
    
    private LocalDateTime updateTime;
    
    private Long adminDeleteTime;

    private Long userDeleteTime;
}

File: hardware-mall-admin/src/api/admin/order.ts
import request from '@/utils/request'

export interface OrderItem {
  id: number
  orderId: number
  productId: number
  productName: string
  productSpec?: string
  productImage?: string
  price: number
  quantity: number
  subtotal: number
}

export interface Order {
  id: number
  orderNo: string
  userId: number
  receiverName?: string
  receiverPhone?: string
  receiverAddress?: string
  totalAmount: number
  freightAmount: number
  payAmount: number
  status: number
  statusText?: string
  logisticsName?: string
  logisticsNo?: string
  deliveryType?: number
  deliveryTypeText?: string
  payTime?: string
  shipTime?: string
  receiveTime?: string
  cancelTime?: string
  cancelReason?: string
  buyerRemark?: string
  createTime?: string
  items?: OrderItem[]
}

export interface OrderQuery {
  page?: number
  limit?: number
  userId?: number
  status?: number
  orderNo?: string
}

export const getOrderList = (params?: OrderQuery) => {
  return request.get<any>('/admin/order/list', { params })
}

export const getOrderById = (id: number) => {
  return request.get<Order>(`/admin/order/${id}`)
}

export const getOrderStats = () => {
  return request.get<any>('/admin/order/stats')
}

export interface ShipOrderPayload {
  deliveryType: number
  logisticsId?: number
}

export const shipOrder = (id: number, payload: ShipOrderPayload) => {
  return request.put(`/admin/order/${id}/ship`, payload)
}

export const refundOrder = (id: number, reason: string) => {
  return request.put(`/admin/order/${id}/refund`, { reason })
}

export const rejectRefund = (id: number, reason: string) => {
  return request.put(`/admin/order/${id}/reject-refund`, { reason })
}
